docs/en/pact-reference.md

```diff
@@ -2671,15 +2671,15 @@ They are used in [with-read](pact-functions.html#with-read), [with-default-read]
 ```
 
 ### Lambdas {#lambdas}
-Lambdas are supported in `let` and `let*`.
-Note: Lambdas are not atoms, since in pact we do not allow arbitrary recursion, therefore
-they must be let-bound and named.
+Lambdas are supported in `let` and `let*`, and as inline function arguments in value positions.
 
 ```lisp
   ; identity function
   (let ((f (lambda (x) x))) (f a))
   ; Higher order example
   (let ((hof (lambda (f a b)  (f a b))))  (hof + 1 2))
```

Comment:
```suggestion
  (let ((hof (lambda (f a b)  (f a b))))  (hof (+) 1 2))
```

Reply:
Other than native callsites (E.g map, filter, zip, fold, fold-db, etc, where we know the function arity beforehand) arbitrary TApps don't really have a way of checking whether a function has been partially applied or not, in an an `App`, other than checking function arity via the type (currently, we just don't return closures as values, they're only syntactic values before reduction via application and not before).

Since `TApp` arguments to functions are simply reduced, making `(f (+ 1 2))` = `(f 3)` before `f` is applied, we'd have to add a check to all apps passed as arguments to user functions for saturation when passed as `(f (+) a)`. 

Moreover, there's one case that's a bit egregious: some function `(defun f () a)` and `(lambda () a)`. In this case `(f)` and `f` have different semantics (one means "run the nullary function" and another is simply the straight reference to it) and thus when passed to some function expecting a nullary function, what does `(g (f) a)` mean? does it mean "reduce the nullary function to its value" or "pass as a nullary function to be called"? 

I can add provisions to checking `TApp` args for saturation but I don't quite think it's worth it. `_` syntax will make it explicit when passing in natives in further pact versions (e.g `(f (+ _ _) a)`) and while It's a bit lame that there's a bit of an inconsistency in partially applying some functions within natives vs user functions, I do think it's worth not adding this machinery (and this will be fixed in core regardless).

Reply:
Does this mean that for e.g. `fold` we'd continue to write `(fold (+) 1 [2 3 4])`, but for a user arg we write `(hof + 1 2)`?

Reply:
HOFs taken out of this release :+1: